app/controllers/concerns/parameters.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Reusable parameter checking for all controllers
+module Parameters
+  extend ActiveSupport::Concern
+
+  ParamType = ActionController::Parameters
+
+  RELATIONSHIP_TYPE = ParamType.map(
+    id: ParamType.string,
+    type: ParamType.string
+  )
```

Comment:
Type validations thus far are for request shape only (i.e. data, attributes, relationships are all `ParamType.map`) and not individual attribute types with the exception of type/id here which holds for any relationship.